Update GzSpinBox to use the SpinBox QML module from QtQuick.Controls2

azeey opened the pull request:
Toward #586

## Summary
The SpinBox module from QtQuick.Controls1 has been deprecated in Qt5 and removed in Qt6. The new SpinBox has a different API and does not directly support floating point values. The workaround in this PR is to store the floating point numbers in a fixed-point representation.

The implementation was inspired by the example in https://doc.qt.io/qt-5/qml-qtquick-controls2-spinbox.html

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

iche033 commented:
code changes look good to me. 

I noticed an issue that when expanded the `-` button overlaps with the display text:

<img width="485" alt="Screenshot 2025-02-05 at 1 49 39 PM" src="https://github.com/user-attachments/assets/b0ed11c9-df24-4f9e-a039-03d34010a31e" />

Also not sure if it's possible to make the `-` and `+` buttons a different background color from the display text field (something like the screenshots in https://doc.qt.io/qt-5/qml-qtquick-controls2-spinbox.html). At first glance, the `-` button makes it look like the spinbox contains a negative number.

azeey commented:
> code changes look good to me.
> 
> I noticed an issue that when expanded the `-` button overlaps with the display text:
> 
> <img alt="Screenshot 2025-02-05 at 1 49 39 PM" width="485" src="https://private-user-images.githubusercontent.com/4000684/410199463-b0ed11c9-df24-4f9e-a039-03d34010a31e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg3OTg3MjMsIm5iZiI6MTczODc5ODQyMywicGF0aCI6Ii80MDAwNjg0LzQxMDE5OTQ2My1iMGVkMTFjOS1kZjI0LTRmOWUtYTAzOS0wM2QzNDAxMGEzMWUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMDVUMjMzMzQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWVlZGEyYTllNjBjMWU1YTc4MzA4MzA3Y2ZhYTUzOWJhZDRiNDJhYmMyZGNhZWU3ODM4MzM4MGE4ZjU0ZTBmYSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.nLvO9Irz_ygTIB8CKA0zD-stpVe68YjkcdMuf306lvk">
> Also not sure if it's possible to make the `-` and `+` buttons a different background color from the display text field (something like the screenshots in https://doc.qt.io/qt-5/qml-qtquick-controls2-spinbox.html). At first glance, the `-` button makes it look like the spinbox contains a negative number.

I've addressed your feedback in https://github.com/gazebosim/gz-gui/pull/654/commits/7330c0655ae88dc8d326d6a6c66ebfa933e5086a and actually got it to look like the SpinBox from the previous version.

![image](https://github.com/user-attachments/assets/c8dc21d0-bfcf-45cf-bc01-8aefc18749cf)


https://github.com/gazebosim/gz-gui/pull/654/commits/8eae434b57da4e8dbf87681b6bf1a769fcb7d67b cleans up some warnings related to bindings.

iche033 reviewed (approved):
nice, works well for me
